### PR TITLE
ravencyan-proposal-submission

### PR DESCRIPTION
**Pull request by ravencyan**
---
A good PR should include the following parts:

- [ ] Provide a clear title
- [ ] Provide meaningful descriptions in the PR and the commit logs
- [ ] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [ ] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [ ] Exclude irrelevant files, such as binaries, text files, or dot files
- [ ] Make sure tests/CI pass

### CODE REVIEW

File: project/ravencyan/README.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Basic Information
+=================
+
+Github Repository: https://github.com/ravencyan/Root-Finding-Algorthms
+
+Problem to Solve & Prospective Users
+====================================
+
+When it comes to finding the solution of equations, root finding algorithms are
```
Comment (yungyuc) [inline project/ravencyan/README.rst:13]: What kind of equations will you solve for?

File: project/ravencyan/README.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+accross different fields, for example, engineers, physicists, computer scientists,
+financial analysts. Among the existing algorithms, the performance in terms of
+conversion rate, computational efficiency and robustness need to be compared and
+verified. In addition, conditions such as initial guesses, convergence criteria,
+and handling special cases need to be taken into consideration.
+
+System Architecture
+===================
+
+The program will provide an prompt for the user to input an equation in the format:
```
Comment (yungyuc) [inline project/ravencyan/README.rst:24]: Please beef up your descriptions of architecture.

File: project/ravencyan/README.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Schedule
+========
+
+* Week 1 (04/08): Search for the root finding algorithms to be implemented in the program, and get a thorough understanding on them.
+* Week 2 (04/15): Implement 2-3 algorithms and perform unit tests to check if they are valid.
+* Week 3 (04/22): Add functions in the algorithms to record their performance for comparison.
+* Week 4 (04/29): Build the input prompt for users to input equations.
+* Week 5 (05/06): Put the algorithms, input prompt, output, and comparison table together into a API.
+* Week 6 (05/13): Add 1-2 more algoirthms. Optimize the codes (all).
+* Week 7 (05/20): Final check for any potential bugs, update the documentation to the final version.
```
Comment (yungyuc) [inline project/ravencyan/README.rst:53]: Debugging should be part of the iterations of implementation.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [ ] You missed an entry in `project/README.rst` as [requested](https://github.com/yungyuc/nsdhw_24sp/blob/567be93bc63a6fb1c2c55b8faf5825172daeaba5/project/README.rst?plain=1#L22)
- [ ] Please add an entry for this PR in E3 (I did not see it)
- [ ] Specify the types or constraints of the equations to solve for.
- [ ] Please beef up your descriptions of architecture.

When you are ready for the next review, leave a comment, clearly write down something like "I am ready, please review again".

**Comment by yungyuc** [comment]
@ravencyan Please update the proposal and E3.

**Comment by yungyuc** [comment]
> @ravencyan Please update the proposal and E3.

@ravencyan This is a reminder that I cannot grade your proposal since the points have not been fully addressed.

**Comment by ravencyan** [comment]
Dear professor,

Sorry I’m late, I will update my proposal as soon as possible!

Best Regards,
吳玟叡

Yung-Yu Chen ***@***.***>於 2024年5月7日 週二，下午10:23寫道：

**Pull request merged**